lib-extra/src/main/java/com/diffplug/spotless/extra/java/EclipseFormatterStep.java

```diff
@@ -47,14 +40,14 @@ private EclipseFormatterStep() {}
 	private static final String FORMATTER_METHOD = "format";
 
 	/** Creates a formatter step for the given version and settings file. */
-	public static FormatterStep create(File settingsFile, Provisioner provisioner) {
-		return create(defaultVersion(), settingsFile, provisioner);
+	public static FormatterStep create(Iterable<File> settingsFiles, Provisioner provisioner) {
```

Comment:
I disagree, but I don't think this issue is a big deal.  Feel free to mark the legacy methods as deprecated.

In general, I'm wary of supporting theoretical usecases without a real example.  Spotless is not flexible enough to allow different line endings on different lines of the same file - e.g. even lines `\n`, odd lines `\r\n`.  If someone wanted to contribute this functionality, I think we would both reject it as too much complexity for too little practical value.

I don't understand the usecase of multiple property files for the eclipse formatter, but arguments are time consuming, your implementation is good and easy-to-read, so I happily give you the benefit of the doubt that this is a feature that at least you will use.

But as this change requires more churn to the existing code it becomes more expensive, and eats into its "benefit of the doubt" grant.  The following are things that make a change more expensive, ranked from most to least:

1. Change user-facing docs
2. Break binary compat for a published lib
3. Deprecate a method
4. Write new code

I've made my point about the costs of this change as best I can.  You understand the benefits of this change better than me, so I trust your judgement in weighing the benefits vs costs, and I'll accept this PR happily (deprecated or not) so long as it doesn't unnecessarily break binary compat.